src/wam2layers/tracking/backtrack.py

```diff
@@ -25,15 +26,15 @@
     write_output,
 )
 from wam2layers.tracking.shared import initialize_tagging_region, initialize_time
-from wam2layers.utils.profiling import ProgressTracker
+from wam2layers.utils.profiling import ProgressTracker, warn_about_gains
 
 logger = logging.getLogger(__name__)
 
 
 def backtrack(
```

Comment:
A large amount of code is shared between the forwardtrack and backwardtrack functions now (13 lines at the start of the functions, gains accounting, output updating...). Seeings as these lines are identical, could this be refactored in a way to reduce the duplication? 

The initialize functions in each module are also nearly identical.

`run_experiment` is also nearly duplicated:

<details><summary>Comparing run_experiment</summary>
<p>

![image](https://github.com/WAM2layers/WAM2layers/assets/12114825/270f7868-9a1f-4c33-bdd1-6fe5ac4c39b7)

</p>
</details> 

I guess it's a balancing act between readability and reducing duplication, so I was wondering what your view on this is.

Reply:
I suggest we make an issue "reduce duplication", but leave it for now

Reply:
I've been working towards this. The fact that they are now already almost the same means we've been quite succesful in aligning them. I agree working towards more dupe reduction is a good goal. Ideally I want the main "run experiment" functions to be something like 

```py
from wam2layers import model

m = model()
m.initialize(config_file)
while model < model.start_time # this is different for forward/backward
    model.update
    if ... write output
model.finalize()
```